src/core/lib/gprpp/thd_posix.cc

```diff
@@ -48,6 +49,23 @@ struct thd_arg {
   bool tracked;
 };
 
+size_t RoundUpToPageSize(size_t size) {
+  size_t page_size = static_cast<size_t>(getpagesize());
```

Comment:
Based on the Linux manual page http://man7.org/linux/man-pages/man2/getpagesize.2.html, we should use sysconf(_SC_PAGESIZE) for compatibility. Also, we do not need to read this value every time when RoundUpToPageSize() is called. It is a constant value unchanged during the lifetime of a process.